apps/web/src/actions/documents/create.ts

```diff
@@ -9,16 +9,14 @@ export const createDocumentVersionAction = authProcedure
   .createServerAction()
   .input(
     z.object({
-      commitUuid: z.string(),
-      documentType: z
-        .enum([
-          'folder' as DocumentType.Folder,
-          'document' as DocumentType.Document,
-        ])
-        .optional(),
       name: z.string(),
+      commitUuid: z.string(),
       parentId: z.number().optional(),
+      documentType: z.nativeEnum(DocumentType).optional(),
     }),
     { type: 'json' },
   )
-  .handler(async ({ input }) => createDocumentVersion(input))
+  .handler(async ({ input }) => {
+    const result = await createDocumentVersion(input)
+    return result.unwrap()
```

Comment:
This was one of the issues. If the server actions return data to the frontend it has to be serializable
```
Error: Only plain objects, and a few built-ins, can be passed to Server Actions. Classes or null prototypes are not supported. 
```